clients/java/signalr/src/main/java/com/microsoft/signalr/InvocationRequest.java

```diff
@@ -17,21 +17,21 @@
 
     public void complete(CompletionMessage completion) {
         if (completion.getResult() != null) {
-            pendingCall.complete(completion.getResult());
+            pendingCall.onSuccess(completion.getResult());
         } else {
-            pendingCall.completeExceptionally(new HubException(completion.getError()));
+            pendingCall.onError(new HubException(completion.getError()));
         }
     }
 
     public void fail(Exception ex) {
-        pendingCall.completeExceptionally(ex);
+        pendingCall.onError(ex);
     }
 
     public void cancel() {
-        pendingCall.cancel(false);
+        pendingCall.onError(new RuntimeException("Invocation was canceled."));
     }
 
-    public CompletableFuture<Object> getPendingCall() {
+    public SingleSubject<Object> getPendingCall() {
```

Comment:
nit: make this return a `Single`. There's no need for the caller to get the subject.